src/main/java/li/zhongfu/cs2103/chatbot/Duke.java

```diff
@@ -0,0 +1,189 @@
+                            } catch (DateTimeParseException e) {
+                                throw new DukeException(String.format("Unknown date format! Try: deadline %s /by 1 jan 2050 12:15", result.getPosArg()));
+                            }
+                        } else if ("event".equals(cmd)) {
+                            if (!result.hasNonblankArg("at")) {
+                                throw new DukeException(String.format("You need a time for your event! Try: event %s /at 1 jan 2050 12:15", result.getPosArg()));
+                            }
+
+                            try {
+                                task = new Event(result.getPosArg(), this.ui.parseDateTime(result.getArg("at")));
+                            } catch (DateTimeParseException e) {
+                                throw new DukeException(String.format("Unknown date format! Try: event %s /at 1 jan 2050 12:15", result.getPosArg()));
+                            }
+                        } else {
+                            task = new ToDo(result.getPosArg());
+                        }
+                        
+                        this.tasks.add(task);
+                        this.ui.printDialog(new String[] {
+                            "New task added:",
+                            String.format(" %s", task),
+                            String.format("You now have %d tasks in your task list.", this.tasks.size())
+                        });
+                        break;
+
+                    case "reload":
+                        try {
+                            this.tasks = TaskList.loadFromStorage(this.storage);
+                            this.ui.printDialog(String.format("Tasks reloaded; %d tasks in list now", this.tasks.size()));
+                        } catch (FileNotFoundException e) {
+                            this.ui.printDialog("Saved tasks file not found! Skipping reload.");
+                        } catch (StorageException e) {
+                            this.ui.printDialog("Error loading tasks! Skipping reload.");
+                        }
+                        break;
+                    
+                    case "save":
+                        this.tasks.save(this.storage);
+                        this.ui.printDialog("Tasks saved!");
+                        break;
+
+                    case "bye":
+                        this.tasks.save(this.storage);
+                        this.ui.printDialog("Bye. Hope to see you again soon!");
+                        return;
+
+                    default:
+                        throw new DukeException("I don't know what that means! Try: todo, deadline, event, list, delete, mark, unmark, reload, save, bye");
+                }
+            } catch (DukeException e) {
+                this.ui.printDialog(e.getMessage().split("\n"));
+            }
+        }
+    }
```

Comment:
According to the code quality guideline, the method is too long.
Source: https://nus-cs2103-ay2122s2.github.io/website/se-book-adapted/chapters/codeQuality.html